tests/python/pants_test/util/test_objects.py

```diff
@@ -320,10 +321,15 @@ class NoFields(datatype()): pass
 
     with self.assertRaises(ValueError) as cm:
       class JustTypeField(datatype([str])): pass
-    expected_msg = (
-      "Type names and field names can only contain alphanumeric characters "
-      "and underscores: \"<type 'str'>\"")
-    self.assertEqual(str(cm.exception), expected_msg)
+    def compare_string_rep(string_type):
```

Comment:
Including `from builtins import str` at the top of this file changes the behavior of these tests under `TypedDatatypeTest`. Instead of returning `str`, `datatype` returns `newstr`, which is `future`'s backport of Python3's `str` class. 

As I understand, this change is introduced only because changes made to this test file, not to the underlying `objects.datatype()` function. When you remove `from builtins import str` from this file, the old version of the tests pass. That's a good thing - we aren't breaking the interface for a bunch of things using `objects.datatype()`.

Reply:
I think your understanding here is correct, in case you were wondering.